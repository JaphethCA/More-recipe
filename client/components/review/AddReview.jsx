import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import handleRecipeReview from '../../actions/requestHandlers/handleRecipeReview';


const propTypes = {
  recipe: PropTypes.objectOf(PropTypes.any).isRequired,
  handleRecipeReview: PropTypes.func.isRequired
};

class AddReview extends Component {
  constructor(props) {
    super(props);
    this.state = {
      content: '',
    };
  }
  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }
  onSubmit = (event) => {
    event.preventDefault();
    this.props.handleRecipeReview(this.props.recipe.id, this.state);
    this.setState({ content: '' });
  }

  render() {
    return (
      <div>
        <div >
          <form onSubmit={this.onSubmit} ref={node => this.reviewForm = node}>
            <div className="input-field" >
              <i className="material-icons prefix">mode_edit</i>
              <textarea
                type="text"
                name="content"
                onChange={this.onChange}
                id="reviewText"
                value={this.state.content}
                className="materialize-textarea"
              />
              <label htmlFor="reviewText">Write a Review</label>
            </div>
            <div className="input-field">
              <button type="submit" className="btn brown" >Post Review</button>
            </div>

          </form>
        </div>
      </div>
    );
  }
}

AddReview.propTypes = propTypes;

export default connect(null, { handleRecipeReview })(AddReview);
