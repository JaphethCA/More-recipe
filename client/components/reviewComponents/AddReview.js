import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import handleRecipeReview from '../../actions/requestHandlers/handleRecipeReview';

class AddReview extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: '',
      content: '',
    }
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
   }
  onChange(e){
    this.setState({[e.target.name]: e.target.value});
  }
  onSubmit(e){
    e.preventDefault();
    this.props.handleRecipeReview(this.props.recipe.id, this.state);
    this.reviewForm.reset();
  }

  render() {
    return (
      <div>
        <div >
          <form onSubmit={this.onSubmit} ref={(node) => this.reviewForm = node }>
            <div className='input-field'>
              <input type='text' name='title' onChange={this.onChange} />
              <label>Title</label>
            </div>
            <div className='input-field' >
              <input type='text' name='content' onChange={this.onChange} />
              <label htmlFor='content'>Review</label>
            </div>
            <div>
              <button type='submit' className='btn blue' >Post Review</button>
            </div>
            
          </form>
        </div>
      </div>
    )
  }
}

AddReview.propTypes = {
  recipe: PropTypes.object.isRequired,
  handleRecipeReview: PropTypes.func.isRequired
}

export default connect(null, {handleRecipeReview})(AddReview);