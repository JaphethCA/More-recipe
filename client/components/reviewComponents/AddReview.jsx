import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import handleRecipeReview from '../../actions/requestHandlers/handleRecipeReview';

class AddReview extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      content: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit(e) {
    e.preventDefault();
    this.props.handleRecipeReview(this.props.recipe.id, this.state);
    this.reviewForm.reset();
  }
                            
  render() {
    return (
      <div>
        <div >
          <form onSubmit={this.onSubmit} ref={node => this.reviewForm = node}>
            <div className="input-field" >
            <i className="material-icons prefix">mode_edit</i>
              <textarea type="text" name="content" onChange={this.onChange} id="reviewText" className="materialize-textarea" />
              <label htmlFor="reviewText">Write a Review</label>
            </div>
            <div className="input-field">
              <button type="submit" className="btn grey" >Post Review</button>
            </div>

          </form>
        </div>
      </div>
    );
  }
}

AddReview.propTypes = {
  recipe: PropTypes.object.isRequired,
  handleRecipeReview: PropTypes.func.isRequired
};

export default connect(null, { handleRecipeReview })(AddReview);
