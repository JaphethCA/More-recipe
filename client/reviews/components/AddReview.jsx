import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {
  content: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

const AddReview = ({ content, onChange, onSubmit }) => (
  <form onSubmit={onSubmit} className="review-form">
    <div className="review-input" >
      <textarea
        type="text"
        name="content"
        onChange={onChange}
        id="reviewText"
        value={content}
        placeholder="add your reviews here"
        className=""
      />
    </div>
    <div >
      <button type="submit" className="btn-class" >Post Review</button>
    </div>
  </form>
);

AddReview.propTypes = propTypes;

export default AddReview;
