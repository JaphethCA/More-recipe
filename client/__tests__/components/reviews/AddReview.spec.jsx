import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import AddReview from '../../../Reviews/components/AddReview';


const props = {
  content: 'new review',
  onChange: jest.fn(),
  onSubmit: jest.fn()
};

describe('<AddReview /> ', () => {
  it('should mount without exploding', () => {
    const wrapper = shallow(<AddReview {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('form').length).toBe(1);
  });

  it('should have a textarea element', () => {
    const wrapper = shallow(<AddReview {...props} />);
    expect(wrapper.find('textarea').length).toBe(1);
    expect(wrapper.find('textarea').parent().is('div')).toBe(true);
  });
  it('should match match snapshot', () => {
    const wrapper = shallow(<AddReview {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

