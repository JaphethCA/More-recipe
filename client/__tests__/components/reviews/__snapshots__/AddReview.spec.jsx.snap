// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddReview />  should match snapshot 1`] = `
<form
  className="review-form"
  onSubmit={[Function]}
>
  <div
    className="review-input"
  >
    <textarea
      className=""
      id="reviewText"
      name="content"
      onChange={[Function]}
      placeholder="add your reviews here"
      type="text"
      value="new review"
    />
  </div>
  <div>
    <button
      className="btn-class"
      disabled={false}
      type="submit"
    >
      Post Review
    </button>
  </div>
</form>
`;
